### PR TITLE
Refactor fed envy event handling

### PR DESCRIPTION
**Pull request by ffranr**
This PR refactors the federation envoy event handling routine with the goal of code simplification and to gain some line length breathing room.

This PR shouldn't change any "business logic".

### CODE REVIEW

File: universe/auto_syncer.go [hunk 1]
```diff
@@ -403,88 +403,18 @@ func (f *FederationEnvoy) syncer() {
...
-					)
-				f.pushProofToFederation(
-					ctxPush, entry.UniID, entry.LeafKey,
-					&entry.Leaf, servers, true,
-				)
-				cancelPush()
+				// Warn, but don't exit the syncer. The syncer
+				// should continue to run and attempt handle
+				// more events.
+				log.Warnf("unable to handle tick event: %v",
```
Comment (guggero) [inline universe/auto_syncer.go:416]: nit: log messages should start capitalized, here and in the other 2 commits.

File: universe/auto_syncer.go [hunk 2]
```diff
@@ -624,6 +563,74 @@ func (f *FederationEnvoy) handleTickEvent() error {
...
+		return fmt.Errorf("nil push request")
+	}
+
+	// First, we'll attempt to registrar the proof leaf with the local
+	// registrar server.
+	ctx, cancel := f.WithCtxQuit()
+	newProof, err := f.cfg.LocalRegistrar.UpsertProofLeaf(
+		ctx, pushReq.ID, pushReq.Key, pushReq.Leaf,
+	)
+	cancel()
```
Comment (guggero) [inline universe/auto_syncer.go:538]: nit: we can use `defer` to cancel these contexts, since we're in a method now.

**Review by guggero — APPROVED** [review]
Nice refactor, LGTM :tada:

File: universe/auto_syncer.go (hunk 2 above)
Comment (ffranr) [reply universe/auto_syncer.go]: I've added fixups which use `defer`. Please let me know if that's what you intended.

**Review by GeorgeTsagk — APPROVED** [review]
LGTM!
Nice cleanup, I'm always a fan of this approach as `syncer` now is simpler & more readable. No need to go through code you don't want to read :+1:

**Review by guggero — APPROVED** [review]

**Pull request merged**